test_suite/tests/derive.rs

```diff
@@ -121,8 +129,14 @@ fn enum_derive() {
 		.type_params(tuple_meta_type!(bool))
 		.variant(
 			Variants::with_fields()
-				.variant("A", Fields::unnamed().field_of::<bool>())
-				.variant("B", Fields::named().field_of::<bool>("b"))
+				.variant(
+					"A",
+					Fields::unnamed().field(Field::unnamed_of::<bool>().with_type_display_name(vec!["T"])),
+				)
+				.variant(
+					"B",
+					Fields::named().field(Field::named_of::<bool>("b").with_type_display_name(vec!["T"])),
+				)
```

Comment:
This was a bit more ergonomic before; having to repeat the info about building a `named`/`unnamed` field is a bit unfortunate. I wonder if `Fields::field(Field::unnamed_of::<bool>().with_type_display_name(…)` could work? 

And what if `Field::unnamed_of` was a free standing function, then it would be just `Fields::field(unnamed_of::<bool> … …)`?

Reply:
Outdated now